dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -3600,8 +3601,17 @@ public void deleteRelatedContent(final Contentlet contentlet, final Relationship
                 .dbRelatedContent(relationship, contentlet, hasParent);
         cons = permissionAPI
                 .filterCollection(cons, PermissionAPI.PERMISSION_READ, respectFrontendRoles, user);
-        APILocator.getRelationshipAPI().deleteByContent(contentlet, relationship, cons);
 
+        for (final Contentlet relatedContent : cons) {
```

Comment:
Does this logic consider non-self related content?

Reply:
The bug not happen for non-self related